ivi-input-modules/ivi-input-controller/src/ivi-input-controller.c

```diff
@@ -1201,28 +1239,26 @@ destroy_input_context(struct input_context *ctx)
     struct ivisurface *tmp_surf_ctx;
     struct wl_resource *resource, *tmp_resource;
 
-    wl_resource_for_each_safe(resource, tmp_resource, &ctx->resource_list) {
-        /*We have set destroy function for this resource.
-         * The below api will call unbind_resource_controller and
-         * free up the controller structure*/
-        wl_resource_destroy(resource);
+    wl_list_for_each_safe(seat, tmp, &ctx->seat_list, seat_node) {
+        destroy_seat(seat);
```

Comment:
we should not destroy seats in destroy_input_context. we will get anyway seat_destroyed signal when backend is destroyed.

Reply:
input controller is being destroyed and after this ideally no more functionality of input controller needs to be invoked. This will lead to complicated handling of the resource which need to freed and which need to be kept (OR leaked). Hence we can do clean up similar to default grab and input.c as explained in my previous comment.